services/arrow-flight/src/test/java/com/dremio/service/flight/AbstractTestFlightSqlServer.java

```diff
@@ -142,4 +144,24 @@ public void testGetTablesFilteringByMultiTableTypes() throws Exception {
       Assert.assertEquals(root.getRowCount(), 0);
     }
   }
+
+  @Test
+  public void testGetTablesTypes() throws Exception {
+    FlightSqlClient flightSqlClient = getFlightClientWrapper().getSqlClient();
+    FlightInfo flightInfo = flightSqlClient.getTableTypes(getCallOptions());
+    try (FlightStream stream = flightSqlClient.getStream(flightInfo.getEndpoints().get(0).getTicket(), getCallOptions())) {
+      Assert.assertTrue(stream.next());
+      VectorSchemaRoot root = stream.getRoot();
+
+      final TableType[] values = TableType.values();
```

Comment:
Can you validate that the values returned here are the same that are returned from getTables?

Reply:
This enum returns more values than what getTables return.

GetTables only return "SYSTEM_TABLE" and in that enum we have 5 types "UNKOWN_TABLE_TYPE", "TABLE", "SYSTEM_TABLE", "VIEW", "UNRECOGNIZED"

Reply:
Ok - so are we returning all 5 as table types from getTableTypes? We should ignore UNKNOWN_TABLE_TYPE and UNRECOGNIZED.

Reply:
Yes, I was returning all of them. I will ignore the  UNKNOW_TABLE_TYPE and UNRECOGNIZED ones